accord-core/src/main/java/accord/local/SerializerSupport.java

```diff
@@ -90,11 +93,80 @@ public class SerializerSupport
                 .addAll(APPLY_TYPES)
                 .build();
 
+    private static final Set<MessageType> BOOTSTRAP_TYPES = ImmutableSet.of(MessageType.BOOTSTRAP_ATTEMPT_MARK_BOOTSTRAP_COMPLETE,
+                                                                            MessageType.BOOTSTRAP_ATTEMPT_COMPLETE_MARKER);
+
+    private static Command localOnly(Agent agent, RangesForEpoch rangesForEpoch, Mutable attrs, SaveStatus status, Timestamp executeAt, @Nullable Timestamp executesAtLeast, Ballot promised, Ballot accepted, WaitingOnProvider waitingOnProvider, MessageProvider messageProvider)
+    {
+        Set<MessageType> witnessed = messageProvider.test(BOOTSTRAP_TYPES);
+        Writes writes = null;
+        Result results = null;
+        if (status.hasBeen(Status.PreApplied))
+        {
+            Invariants.checkArgument(witnessed.contains(MessageType.BOOTSTRAP_ATTEMPT_MARK_BOOTSTRAP_COMPLETE), "%s LocalOnly command was missing BOOTSTRAP_ATTEMPT_MARK_BOOTSTRAP_COMPLETE; found %s", status, new LoggedMessageProvider(messageProvider));
+            var marker = messageProvider.bootstrapAttemptMarkBootstrapComplete();
+            Txn emptyTxn = agent.emptyTxn(marker.localSyncId.kind(), marker.ranges);
+            writes = emptyTxn.execute(marker.localSyncId, marker.localSyncId, null);
+            results = emptyTxn.result(marker.localSyncId, marker.localSyncId, null);
+            Ranges coordinateRanges = rangesForEpoch.coordinates(attrs.txnId());
+            attrs.partialTxn(emptyTxn.slice(coordinateRanges, true))
+                 .partialDeps(Deps.NONE.slice(coordinateRanges));
+        }
+        else
+        {
+            Invariants.checkArgument(witnessed.contains(MessageType.BOOTSTRAP_ATTEMPT_COMPLETE_MARKER), "%s LocalOnly command was missing BOOTSTRAP_ATTEMPT_COMPLETE_MARKER; found %s", status, new LoggedMessageProvider(messageProvider));
+            var marker = messageProvider.bootstrapAttemptCompleteMarker();
+            var keys = marker.valid;
+            var route = marker.syncPoint.route();
+
+            Txn emptyTxn = agent.emptyTxn(marker.localSyncId.kind(), keys);
+            Ranges coordinateRanges = rangesForEpoch.coordinates(attrs.txnId());
+            attrs.partialTxn(emptyTxn.slice(coordinateRanges, true))
+                 .partialDeps(Deps.NONE.slice(coordinateRanges));
+        }
+
+        switch (status.status)
```

Comment:
honestly several of these status are not valid... the one thing is I don't know if `ProgressLog` is expected to ignore `LocalOnly` or not, so not sure if I can assume the 2 live status expected or not... so went with duplication of code to be safer...

Reply:
Yes, `ProgressLog` should not interact with `LocalOnly` transactions.